### ISSUE TITLE
Can't use multiple async iterators for http requests

### ISSUE THREAD

**Issue by denyncrawford** [denoland/deno#9074]
Hi, I have a problem related to the way the http std server handles connection events.

I am working on a websockets framework to create realtime applications with event-based websocket events, like socket.io (you can find the project as Dropper in the third parties). I am using the std http server to create my Wesockets server with std WS module. Up to this point everything is fine, but I want to extend the functionality of my module by allowing users to pass their own server. For example, if someone already has a server set up or a framework like Opine I would like to be able to pass the user http server and work with it. 

The problem is that I can't use multiple async iterators on a single app, and I understand why. But it makes impossible for me to listen to the external server requests from inside my module if an async iterator already exists.

**My question is: Is there a way to intercept requests from a server that already has an asynciterator?**

This is an example of what I want to do:

```javascript
import { serve } from 'https://deno.land/std@0.83.0/http/server.ts'
import { Server as Dropper } from 'https://deno.land/x/dropper/src/mod.ts'

const server = serve('localhost:8080');
const dropper = new Dropper(server)

for await (const req of server) {
  //foo...
}
```

The functionality is not implemented in Dropper yet, but I have something like this minimal example:

```ts
class Server extends EventEmitter {
  constructor(public server: object) {
   this.server = server || serve('localhost:8080');
   this.init()
  }
  init() {
    const server = this.server;
     for await (const req of server) {
       const {
         headers,
         conn
       } = req;
       acceptWebSocket(/*------*/)
     }
  }
}
```
Basically, if the server is not provided in the instance, it creates a new one. Works great, but if server is provided it throws an error.

**This is the console output**:

```bash
error: Uncaught (in promise) Busy: Another accept task is ongoing
    at processResponse (core.js:223:11)
    at Object.jsonOpAsync (core.js:240:12)
    at async Listener.accept (deno:cli/rt/30_net.js:117:19)
    at async Server.acceptConnAndIterateHttpRequests (server.ts:228:14)
    at async MuxAsyncIterator.callIteratorNext (mux_async_iterator.ts:30:31)
```

I'm open to any idea. Thanks.

**Comment by nayeemrmn** [comment]
In what scenario would the user have attached an external loop to the server before passing it to your library? Are you actually trying to support middleware-ish use cases (it doesn't seem like it since the pre-created server is optional), or do you just want better error messages?

**Comment by denyncrawford** [comment]
@nayeemrmn What I was trying to do is use a server that a user has previously created to handle the websocket requests, this is optional because dropper has a default server. The purpose is that if a user wants to use dropper but already has a server, he does not have to create two different servers in the same process, since it is blocking,.

I solved this by making a function to handle the request and it is separate from the built-in server. Actually, the function that was executed in the server is now a middleware function and you can use it externally, either in a std http server or in a server framework like opine.

## This is how you use Dropper server now: 

### Stand Alone:

```javascript
const options;

const dropper = new Server(options); // Creates a default server on port 8080

dropper.on('connection', socket => { // W
  socket.send('pizza', 'I sent you a pizza!')
  socket.on('thanks', data => {
    console.log(data) // => Thanks btw
  })
})

// Listening for global events

dropper.on('thanks', data => {
  console.log(data) // => Thanks btw
})
```

### With Opine (or any middleware framework):

```javascript
import { opine } from 'https://opinesource/mod.ts'
const app = opine();
import { Server as Dropper } from '...'

const dropper = new Dropper({
  serve: false // Important
})

app.use('/dropper' (req) => dropper.handle(req)); // Don't respond or call next, just use the request.

app.get('/', (req, res) => {
  //...foo
})

dropper.on('connection', socket => {
  ..foo
})

app.listen(3000)
```

### With the std http server:

```javascript
import { serve } from 'https://serverhttpsouce/mod.ts'
import { Server as Dropper } from '...'

const server = serve('localhost:3000')

const dropper = new Dropper({
  serve: false // Important
})

dropper.on('connection', socket => {
  ..foo
})

for await (const req of server) {
  if (req.url === '/dropper') {
    dropper.handle(req);
  } else {
    // return static content, etc
  }
}
``` 

I close this issue even though it is impossible to reuse a server, but I assume that the feasible solution (in my case) is to middlewarize when you need to get data from a request. Thanks!
